python/IECore/MenuDefinition.py

```diff
@@ -153,22 +153,43 @@ def items( self ) :
 	## Returns a new MenuDefinition containing only the menu items
 	# that reside below the specified root path. The paths in this
 	# new definition are all adjusted to be relative to the requested
-	# root.
+	# root. This also takes into account given `subMenu` definitions,
+	# although the value for subMenu can't be a callable for performance reasons.
 	def reRooted( self, root ) :
 
 		if not root:
 			return MenuDefinition( [] )
 
-		root = root + "/" if not root.endswith("/") else root
+		rootConformed = "/{}/".format(root.strip('/'))
 
 		newItems = []
 		for path, itemDict in self.items() :
-
-			if path.startswith( root ) :
-				newItems.append( ( path[len(root)-1:], itemDict ) )
+			if path.startswith( rootConformed ) :
+				newItems.append( ( path[ len( rootConformed )-1 : ], itemDict ) )
+			elif path == rootConformed[:-1] and isinstance( itemDict.subMenu, MenuDefinition ):
+				# NOTE: subMenu values that are `callable` won't be considered here for performance reasons
+				newItems += itemDict.subMenu.items()
 
 		return MenuDefinition( newItems )
 
+	## Returns the item at `path`. Supports multiple path components.
+	# If no item is found, return `None`
+	def getItem( self, searchPath ):
+		strippedPath = searchPath.strip('/')
+
+		# single component path
+		for path, item in self.__items:
+			if strippedPath == path.strip('/'):
+				return item
+
+		# multi component path, e.g. `my/item/path`
+		if '/' in strippedPath:
+			rootPath, _, childPath = strippedPath.partition('/')
+			rootedDef = self.reRooted(rootPath)
+			if rootedDef.size():
+				return rootedDef.getItem(childPath)
```

Comment:
For the common case of a path not involving submenus, it seems a pity to do all this recursive rerooting when we could find the item with the initial search anyway. I'm wondering if this could be formulated to take that into account, and therefore not require the `reRooted()` change...

Reply:
If the path is simple (one component) the lookup will exit as early as possible. The problem is that I want to support all kinds of possible menu definitions, either defining subitems via `multiple/component/path` or passing another definition in `subMenu`. This makes the definitions a lot cleaner and more structured.

Reply:
Ah, I think I misread the code. I took the "single component path" comment at face value and thought it meant "/oneOnly" and not "/one/or/more". Am I correct now in thinking that when you say "multi component" you mean "one with a submenu"?

Reply:
Yes and no, a single component for me would be `/myItem`. Something like `/submenu/myItem` would be a multi component path for me and can include a `subMenu` definition object I would have to reroot in order to get the actual item I'm looking for.